stake-pool/js/src/index.ts

```diff
@@ -505,3 +578,331 @@ export async function withdrawSol(
+        stakeAccountAddress: stakeAccountAddress.toBase58(),
+        validatorActiveStakeLamports: validator.activeStakeLamports.toString(),
+        validatorLastUpdateEpoch: validator.lastUpdateEpoch.toString(),
+        validatorLamports: validator.activeStakeLamports
+          .add(validator.transientStakeLamports)
+          .toString(),
+        validatorTransientStakeAccountAddress: transientStakeAccountAddress.toBase58(),
+        validatorTransientStakeLamports: validator.transientStakeLamports.toString(),
+        updateRequired,
+      };
+    }),
+  );
+
+  const totalPoolTokens = lamportsToSol(stakePool.account.data.poolTokenSupply);
+  const updateRequired = !lastUpdateEpoch.eqn(epochInfo.epoch);
+
+  return {
+    address: stakePoolAddress.toBase58(),
+    poolWithdrawAuthority: withdrawAuthority.toBase58(),
+    manager: stakePool.account.data.manager.toBase58(),
+    staker: stakePool.account.data.staker.toBase58(),
+    stakeDepositAuthority: stakePool.account.data.stakeDepositAuthority.toBase58(),
+    stakeWithdrawBumpSeed: stakePool.account.data.stakeWithdrawBumpSeed,
+    maxValidators: maxNumberOfValidators,
+    validatorList: validatorList.account.data.validators.map(validator => {
+      return {
+        activeStakeLamports: validator.activeStakeLamports.toString(),
+        transientStakeLamports: validator.transientStakeLamports.toString(),
+        lastUpdateEpoch: validator.lastUpdateEpoch.toString(),
+        transientSeedSuffixStart: validator.transientSeedSuffixStart.toString(),
+        transientSeedSuffixEnd: validator.transientSeedSuffixEnd.toString(),
+        status: validator.status.toString(),
+        voteAccountAddress: validator.voteAccountAddress.toString(),
+      };
+    }), // CliStakePoolValidator
+    validatorListStorageAccount: stakePool.account.data.validatorList.toBase58(),
+    reserveStake: stakePool.account.data.reserveStake.toBase58(),
+    poolMint: stakePool.account.data.poolMint.toBase58(),
+    managerFeeAccount: stakePool.account.data.managerFeeAccount.toBase58(),
+    tokenProgramId: stakePool.account.data.tokenProgramId.toBase58(),
+    totalLamports: stakePool.account.data.totalLamports.toString(),
+    poolTokenSupply: stakePool.account.data.poolTokenSupply.toString(),
+    lastUpdateEpoch: stakePool.account.data.lastUpdateEpoch.toString(),
+    lockup: stakePool.account.data.lockup, // pub lockup: CliStakePoolLockup
+    epochFee: stakePool.account.data.epochFee,
+    nextEpochFee: stakePool.account.data.nextEpochFee,
+    preferredDepositValidatorVoteAddress:
+      stakePool.account.data.preferredDepositValidatorVoteAddress,
+    preferredWithdrawValidatorVoteAddress:
+      stakePool.account.data.preferredWithdrawValidatorVoteAddress,
+    stakeDepositFee: stakePool.account.data.stakeDepositFee,
+    stakeWithdrawalFee: stakePool.account.data.stakeWithdrawalFee,
+    // CliStakePool the same
+    nextStakeWithdrawalFee: stakePool.account.data.nextSolWithdrawalFee,
```

Comment:
These mistakes keep coming up in the CLI :sweat: 
```suggestion
    nextStakeWithdrawalFee: stakePool.account.data.nextStakeWithdrawalFee,
```